docs/advanced-features.md

```diff
@@ -0,0 +1,308 @@
+
+Below is an example that demonstrates how to retrieve a user from a database, validate the user state, and then fetch additional authentication data. In this example, a missing user or an archived user short-circuits the computation:
+
+```scala
+import cats.Monad
+import pl.iterators.sealedmonad.syntax._
+import pl.iterators.sealedmonad.examples.{User, AuthMethod, LoginResponse}
+import cats.effect.IO
+
+object UserService {
+  def sealedLogin[M[_]: Monad](
+      email: String,
+      // Simulated functions returning M[Option[_]] from the database
+      findUser: String => M[Option[User]],
+      findAuthMethod: (Long, String) => M[Option[AuthMethod]],
+      issueTokenFor: User => String,
+      checkAuthMethod: AuthMethod => Boolean,
+      authMethodFromUserId: Long => AuthMethod,
+      mergeAccountsAction: (AuthMethod, User) => M[LoginResponse]
+  ): M[LoginResponse] = {
+    val s = for {
+      user <- findUser(email)
+        .valueOr(LoginResponse.InvalidCredentials)
+        .ensure(!_.archived, LoginResponse.Deleted)
+      userAuthMethod = authMethodFromUserId(user.id)
+      authMethod <- findAuthMethod(user.id, userAuthMethod.providerName)
+        .valueOrF(mergeAccountsAction(userAuthMethod, user))
+    } yield {
+      if (checkAuthMethod(authMethod))
+        LoginResponse.LoggedIn(issueTokenFor(user))
+      else
+        LoginResponse.InvalidCredentials
+    }
+    s.run
+  }
+}
+
+// Models used in the example
+package pl.iterators.sealedmonad.examples
+
+trait User {
+  def id: Long
+  def archived: Boolean
+}
+
+trait AuthMethod {
+  def providerName: String
+}
+
+sealed trait LoginResponse
+object LoginResponse {
+  final case class LoggedIn(token: String) extends LoginResponse
+  case object InvalidCredentials         extends LoginResponse
+  case object Deleted                    extends LoginResponse
+}
+```
+
+In the above code:
+- The helper methods **valueOr** and **ensure** capture common error handling patterns.
+- The entire operation is expressed within a for-comprehension, making the logic easy to follow.
+- When a value is missing (or the condition fails) the computation "short-circuits" and returns the appropriate error response.
+
+---
+
+## Composing with Other Monadic Structures
+
+Sealed Monad is designed to work effortlessly with other monadic abstractions such as IO, OptionT, and EitherT from libraries like Cats. Below are examples of such integrations.
+
+### Example: Integrating with IO
+
+When working with Cats Effect's IO, you can lift IO values into the Sealed Monad context using the **seal** and **liftF** methods.
+
+```scala
+import cats.effect.IO
+import pl.iterators.sealedmonad.syntax._
+import pl.iterators.sealedmonad.examples._
+import cats.Monad
+
+object AuthServiceIO {
+  // A helper to lift a pure IO value into the Sealed Monad context
+  def loginWithIO(
```

Comment:
This example is a copy paste of the previous one, the only change it uses concrete monad instead of TF-style.